### PR TITLE
add support for building samba-server on ubi9

### PR DESCRIPTION
**Pull request by andrewschoen**
The podman build command must add the secrets with --secret and the repos in /repos should be mounted in /etc/yum.repos.d when building. The actual rpms should be downloaded from artifactory and mounted to /tmp/samba-build and /tmp/sambacc-build during the build.

### CODE REVIEW

**Comment by andrewschoen** [comment]
I still need to add support for SAMBACC_VERSION_SUFFIX and SAMBA_VERSION_SUFFIX.  I've hardcoded sambacc for now and samba installs fine without the suffix if there is only one version in the repo.

File: Containerfile.rhel [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+LABEL org.opencontainers.image.title="Samba container"
+LABEL org.opencontainers.image.description="Samba container"
+LABEL org.opencontainers.image.vendor="Samba in Kubernetes"
+LABEL org.opencontainers.image.url="https://github.com/samba-in-kubernetes/samba-container"
+
+
+COPY smb.conf /etc/samba/smb.conf
+COPY --chmod=755 install-packages-rhel.sh /usr/local/bin/install-packages.sh
+RUN echo "installing packages..."
+RUN --mount=type=secret,id=org-id --mount=type=secret,id=act-key /usr/local/bin/install-packages.sh \
```
Comment (phlogistonjohn) [inline Containerfile.rhel]: I find `act-key` a bit confusing. I'm assuming this is short for "account-key"? If so, I think we should just spell it out as `account-key` as "account" is not very long, and typically I see it abbreviated as "acct" and "act" is already a common english word.

File: Containerfile.rhel [hunk 2]
```diff
@@ -0,0 +1,36 @@
+FROM --platform=${BUILDPLATFORM:-linux/amd64} registry.access.redhat.com/ubi9
```
Comment (phlogistonjohn) [inline Containerfile.rhel]: What does the `--platform` switch accomplish? (I did review the docs so I know what it does.)

File: Containerfile.rhel (hunk 1 above)
Comment (andrewschoen) [reply Containerfile.rhel]: That makes sense and you are right, it's short for "account-key". I can make that change.

File: Containerfile.rhel (hunk 2 above)
Comment (andrewschoen) [reply Containerfile.rhel]: It's for eventual multi-arch support. I had looked into it and added this, we only have amd64 rpms right now though anyway.

Comment (andrewschoen) [reply Containerfile.rhel]: Considering that, I'll just remove it for now. I can add it back in if/when we need to.

File: Containerfile.rhel (hunk 1 above)
Comment (andrewschoen) [reply Containerfile.rhel]: I was wrong, it actually stands for "activation-key". I've changed it to that instead.

**Review by phlogistonjohn — COMMENTED** [review]
looks ok to me at this point, but I don't currently have visibility into the actual build so I'll shy away from giving formal approval.

**Comment by andrewschoen** [comment]
This PR is being used in https://github.ibm.com/ceph/ceph-pipeline-configs/pull/12

**Pull request merged**